rules/enforce-store-naming-convention/enforce-store-naming-convention.js

```diff
@@ -74,22 +94,35 @@ module.exports = {
           }
 
           const storeName = node.parent.id.name;
-          if (storeName.startsWith("$")) {
+
+          if (mode === "prefix" && storeName.startsWith("$")) {
             continue;
           }
 
+          if (mode === "postfix" && storeName.endsWith("$")) {
+            continue;
+          }
+
+          const correctedStoreName = mode === "prefix"
+          ? `$${storeName}`
+          : `${storeName}$`;
+
           reportStoreNameConventionViolation({
             context,
             node: node.parent,
             storeName,
+            correctedStoreName,
+            mode
           });
           return;
         }
 
         // Store creation with .map
         if (node.callee?.property?.name === "map") {
-          const objectIsEffectorStore =
-            node.callee?.object?.name?.startsWith?.("$");
+          const objectIsEffectorStore = mode === "prefix"
```

Comment:
As a mention before, this logic can be extracted to a separate function, so this piece of code could be simplified.